### ISSUE TITLE
aqc synthesis plugin seems to ignore approximation_degree

### ISSUE THREAD

**Issue by 1ucian0** [Qiskit/qiskit-terra#10386]
The unitary synthesis plugin `aqc` gives always the same result, independently of `approximation_degree`.

Consider the following example:
```
# create a random unitary
from qiskit import QuantumCircuit
from qiskit.quantum_info import Operator
from qiskit.quantum_info.random import random_unitary

qc = QuantumCircuit(3)

qc.unitary(random_unitary(8, seed=0), [0,1,2])

# synthesis (with approximation_degree=0), back to matrix, and sum all the elements 

from qiskit.transpiler.passes.synthesis.unitary_synthesis import UnitarySynthesis
from qiskit.algorithms.optimizers import SLSQP
import numpy as np

result = UnitarySynthesis(basis_gates=["rx", "ry", "rz", "cx"], method='aqc', approximation_degree=0, plugin_config={"seed": 42})(qc)
approx_matrix = Operator(result).data

print(np.sum(approx_matrix))
```
```
(0.9471489692201448-0.04904005347947815j)
```
The result is the same with `approximation_degree=1`

result = UnitarySynthesis(basis_gates=["rx", "ry", "rz", "cx"], method='aqc', approximation_degree=1, plugin_config={"seed": 42})(qc)
approx_matrix = Operator(result).data

print(np.sum(approx_matrix))
```
```
(0.9471489692201448-0.04904005347947815j)
```

It would be great if the approximation_degree could affect the `maxiter` of the optimizer, for example (I'm not an expert in AQC).

For example, the following (with `"optimizer": SLSQP(10)`) takes 1.49s in my laptop.
```
qc = QuantumCircuit(4)
qc.unitary(random_unitary(16, seed=0), [0,1,2,3])

result = UnitarySynthesis(basis_gates=["rx", "ry", "rz", "cx"], method='aqc', plugin_config={"seed": 42, "optimizer": SLSQP(10)})(qc)
```
while the default one (I think is 100) takes 1m 53.1s

**TL;DR** Change default `maxiter` based on the approximation level, if not set. Maybe the amount of unitary boxes and qubits involved should play a role too.

**Comment by jakelishman** [comment]
That said, it probably shoudl be clearer in the docstring of `UnitarySynthesis` that ``approximation_degree`` isn't part of the plugin interface, and only has meaning for the default synthesis (it only still exists at all for backwards compatibility).

**Comment by jakelishman** [comment]
Well, Luciano might disagree with me (I'm not a dictator), but it's probably best to wait for his response at least.

**Comment by SamD-1998** [comment]
As an aside: Is there an issue out there like this one which involves concepts of QM implemented in QC? Eg here was the adiabatic quantum computation and the adiabatic theorem. I find these interesting to work on!

**Comment by SamD-1998** [comment]
Gotcha! Even if they're transpiler plugins, that's fine with me! AQC sort of things just make things more interesting to me.


Anyway, I'll wait for @1ucian0 to respond.
